controllers/utils/mapper.go

```diff
@@ -92,3 +99,97 @@ func NHCByRemediationCRMapperFunc(logger logr.Logger) handler.MapFunc {
 	}
 	return delegate
 }
+
+// MHCByNodeMapperFunc return the Node-to-MHC mapper function
+func MHCByNodeMapperFunc(c client.Client, logger logr.Logger) handler.MapFunc {
+	delegate := func(o client.Object) []reconcile.Request {
+		requests := make([]reconcile.Request, 0)
+
+		node := &v1.Node{}
+		if err := c.Get(context.Background(), client.ObjectKey{Name: o.GetName()}, node); err != nil {
+			if !errors.IsNotFound(err) {
+				node.Name = o.GetName()
+				logger.Info("mapping deleted node", "node name", o.GetName())
+			} else {
+				logger.Error(err, "failed to get node", "node name", o.GetName())
+				return requests
+			}
+		}
+
+		machine, err := getMachineFromNode(c, node.Name)
+		if machine == nil || err != nil {
+			logger.Info("No-op: Unable to retrieve machine from node %q: %v", node.Name, err)
```

Comment:
IIUC (based on getMachineFromNode implementation)  there shouldn't be a use case where machine == nil && err == nil.
Since we can safely assume an error occured, maybe switch from `logger.Info` to `logger.Error`

Reply:
copy/paste from MAO, but you are correct 👍🏼